src/flow/workflow/ICSyncWorkflow.php

```diff
@@ -149,10 +163,36 @@ protected function runRevisionSync($extra_diff_args = array()) {
     $grafted_parent_branches = array();
 
     foreach ($graph->getNodesInTopologicalOrder() as $branch_name) {
+      // generally master branch is not that interesting, skip it
+      if ($branch_name == 'master') {
+        continue;
+      }
       $feature = $this->getFeature($branch_name);
 
-      if (!$feature || !$feature->getAuthorPHID()) {
-        // no revision or author
+      if (!$feature) {
+        $confirm = $this->consoleConfirm(tsprintf(
+          'Branch "%s" has no Differential Revision attached, do you want to '.
+          'create one?', $branch_name));
+        if (!$confirm) {
+          $this->writeInfo(pht(
+            'Skipping branch "%s"', $branch_name), '');
+          continue;
+        }
+        // looks like branch has no Revision, create one and start over again
+        echo "\n";
+        $this->writeInfo(pht(
+          'Creating new Differential Revision for branch "%s"',
+          $branch_name), '');
+        $this->checkoutBranch($branch_name, true);
+        $this->runDiffWorkflow($extra_diff_args);
+        $this->clearFlowWorkspace();
+        $this->checkoutBranch($initial_branch);
```

Comment:
During testing I found that this will not work, `arc diff` is slightly rewriting commit message hence after diff creation I need to cascade changes